File: ExamPortal-main/ExamProtable/src/main/java/com/example/examprotable/service/QuestionService.java
package com.example.examprotable.service;

import com.example.examprotable.entities.exam.Question;
import com.example.examprotable.entities.exam.Quiz;

import java.util.Set;

public interface QuestionService {
    public Question addQuestion(Question question);

    public Question updateQuestion(Question question);

    public Set<Question> getQuestions();

    public Question getQuestion(Long quesid);

    public Set<Question> getQuestionsOfQuiz(Quiz quiz);

    public void deleteQuestion(Long quesid);
}

File: ExamPortal-main/ExamPortal-Frontend/src/app/services/question.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
const baseUrl = 'http://localhost:8080/question';
@Injectable({
  providedIn: 'root',
})
export class QuestionService {
  constructor(private http: HttpClient) {}
  public getQuestionsOfQuiz(qid: any) {
    return this.http.get(`${baseUrl}/quiz/all/${qid}`);
  }
  public getQuestionsOfQuizForUser(qid: any) {
    return this.http.get(`${baseUrl}/quiz/${qid}`);
  }
  public addQuestion(data: any) {
    return this.http.post(`${baseUrl}/`, data);
  }
  public deleteQuestion(quesId: any) {
    return this.http.delete(`${baseUrl}/${quesId}`);
  }
  public getQuestion(quesId:any){
    return this.http.get(`${baseUrl}/${quesId}`);
  }
  public updateQuestion(data: any) {
    return this.http.put(`${baseUrl}/`, data);
  }
}
